src/baldr/datetime.cc

```diff
@@ -719,18 +488,19 @@ tz_db_t::tz_db_t() {
 
   zones.reserve(tz_name_to_id.size());
   for (const auto& tz_pair : tz_name_to_id) {
-    // we find either the official timezone or we get the target timezone of a link
+    // we find either the official timezone or we get the target timezone of a link (i.e. deprecated
+    // time zone)
     auto* tz = db.locate_zone(tz_pair.first);
     zones[tz_pair.second] = &*tz;
   }
 }
 
 size_t tz_db_t::to_index(const std::string& zone) const {
   auto it = tz_name_to_id.find(zone);
-  if (it != tz_name_to_id.cend() || (it = tz_new_to_old_id.find(zone)) != tz_new_to_old_id.end()) {
+  if (it != tz_name_to_id.cend()) {
     return it->second;
   }
-  return 0;
+  throw std::runtime_error(zone + " can't be resolved to a non-deprecated time zone.");
```

Comment:
we should really never have a 0 in the tiles. that can't really happen for our purposes with the tz submodule and the tz shapefile. so if it happens we should throw, bcs then smth is wrong. FWIW, I did check that for the full OSM planet last year, the tz coverage is absolute as should be expected.